packages/grid/_modules_/grid/hooks/features/useGridComponents.tsx

```diff
@@ -1,8 +1,72 @@
 import * as React from 'react';
+import MUICheckbox from '@material-ui/core/Checkbox';
 import { GridApiRef } from '../../models/api/gridApiRef';
 import { GridApiRefComponentsProperty } from '../../models/api/gridComponentsApi';
-import { DEFAULT_GRID_SLOTS_COMPONENTS, GridSlotsComponent } from '../../models/gridSlotsComponent';
+import { GridSlotsComponent } from '../../models/gridSlotsComponent';
 import { GridSlotsComponentsProps } from '../../models/gridSlotsComponentsProps';
+import { GridIconSlotsComponent } from '../../models';
+import {
+  GridArrowDownwardIcon,
+  GridArrowUpwardIcon,
+  GridCheckIcon,
+  GridCloseIcon,
+  GridColumnIcon,
+  GridColumnMenu,
+  GridColumnsPanel,
+  GridFilterAltIcon,
+  GridFilterListIcon,
+  GridFilterPanel,
+  GridFooter,
+  GridHeader,
+  GridLoadingOverlay,
+  GridNoRowsOverlay,
+  GridPagination,
+  GridPanel,
+  GridPreferencesPanel,
+  GridSaveAltIcon,
+  GridSeparatorIcon,
+  GridTableRowsIcon,
+  GridTripleDotsVerticalIcon,
+  GridViewHeadlineIcon,
+  GridViewStreamIcon,
+} from '../../components';
+import { GridColumnUnsortedIcon } from '../../components/columnHeaders/GridColumnUnsortedIcon';
+import { ErrorOverlay } from '../../components/ErrorOverlay';
+import { GridNoResultsOverlay } from '../../components/GridNoResultsOverlay';
+
+const DEFAULT_GRID_SLOTS_ICONS: GridIconSlotsComponent = {
```

Comment:
I had to move the `DEFAULT_GRID_SLOTS_ICONS` outside of `gridSlotsComponent.ts`
Otherwise, any component listed here could not use the context due to circular import.